### PR TITLE
Get Companies responsive

### PR DESCRIPTION
**Pull request by Caleb-Nieto**
### Feature / Bug Description
This feature presents companies data

### Solution
Components created: CompaniesTable, CompanyCard, PaginatedList.
Service created: CompanyService with getCompanies mutation
Add use-debounce dependency to manage the table resize. this is calculated in the tableHeight.ts file.

### Notes
Check it out the design, the card design and the table design are not the best so we can change it in the future of course
Important! The Strings file i dont know why is not working well I hope when I push to the repository it fix it

### Tickets
[WMA-28](https://cdentalcaregroup.atlassian.net/browse/WMA-28)

### Screenshots / Screencasts
dropdown
ipad md
lg
phone
Tarjetas  TPM

[WMA-28]: https://cdentalcaregroup.atlassian.net/browse/WMA-28?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by Caleb-Nieto** [comment]
Forget the last Image

File: src/pages/company/Companies.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import CustomButton from "../../components/CustomButtons";
+import Strings from "../../utils/Strings";
+import CompanyCard from "./components/CompanyCard";
+import PaginatedList from "../../components/PaginatedList";
+
+const Companies = () => {
+  const [getCompanies] = useGetCompaniesMutation();
+  const [data, setData] = useState<Company[]>([]);
+  const [isLoading, setLoading] = useState(false);
+  const [querySearch, setQuerySearch] = useState("");
```
Comment (imadiaz) [inline src/pages/company/Companies.tsx:16]: you can use here the EMPTY string

File: src/pages/company/Companies.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    setLoading(true);
+    const response = await getCompanies().unwrap();
+    setData(response);
+    setDataBackup(response);
+    setLoading(false);
+  };
+
+  useEffect(() => {
+    handleGetCompanies();
+    console.log(data);
```
Comment (imadiaz) [inline src/pages/company/Companies.tsx:34]: do we need thsi log?

File: src/pages/company/Companies.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  }, []);
+
+  const handleSearch = (event: any) => {
```
Comment (imadiaz) [inline src/pages/company/Companies.tsx:37]: in the react we normally use "handleOnSearch" we add the "On" for each handler

File: src/pages/company/components/CompaniesTable.tsx [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+type Sorts = GetSingle<Parameters<OnChange>[2]>;
+
+interface CompaniesTableProps {
+  data: Company[];
+  isLoading: boolean;
+  clearFilters: boolean;
+  setClearFilters: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const CompaniesTable = ({
```
Comment (imadiaz) [inline src/pages/company/components/CompaniesTable.tsx:23]: if we just named like "CompanyTable"?

File: src/utils/Constants.ts [hunk 5]
```diff
@@ -1,7 +1,9 @@
-const EMPTY_STRING = ''
-const WELCOME = 'Welcome!'
+const EMPTY_STRING = "";
```
Comment (imadiaz) [inline src/utils/Constants.ts:1]: you can just use EMPTY you don't need to add the "STRING" word

**Review by imadiaz — APPROVED** [review]
Awesome job heree!! looks pretty cool the code, just give you some comments!

File: src/utils/Constants.ts (hunk 5 above)
Comment (Caleb-Nieto) [reply src/utils/Constants.ts:1]: okay Thank u @imadiaz I was just waiting to change this to the strings files as you mentioned in one ticket can I do that in this next commit?

File: src/pages/company/components/CompaniesTable.tsx (hunk 4 above)
Comment (Caleb-Nieto) [reply src/pages/company/components/CompaniesTable.tsx:23]: Okay its better

File: src/pages/company/Companies.tsx (hunk 2 above)
Comment (Caleb-Nieto) [reply src/pages/company/Companies.tsx:34]: not really

File: src/pages/company/Companies.tsx (hunk 3 above)
Comment (Caleb-Nieto) [reply src/pages/company/Companies.tsx:37]: okay got it

File: src/utils/Constants.ts (hunk 5 above)
Comment (imadiaz) [reply src/utils/Constants.ts:1]: yes

**Pull request merged**